mammoth/utils/parse.py

```diff
@@ -272,6 +270,45 @@ def update_model_opts(cls, model_opts):
             model_opts.lambda_align = 0.0
             model_opts.full_context_alignment = False
 
+    @classmethod
+    def validate_x_transformers_opts(cls, opts):
+        if not opts.x_transformers:
+            opts.x_transformers = dict()
+            return
+        opts_dict = yaml.safe_load(opts.x_transformers)
+        for overwritten_key in (
+            'dim',
+            'depth',
+            'heads',
+            'causal',
+            'cross_attend',
+            'pre_norm_has_final_norm',
+            'ff_mult',
+        ):
+            if overwritten_key in opts_dict:
+                raise ValueError(
+                    f'"{overwritten_key}" is overwritten with values from other Mammoth arguments. '
+                    'You can not set it as part of x_transformers opts.'
+                )
+        for unsupported_key in (
+            'sandwich_coef',    # Sandwich would be very unintuitive with multiple layerstacks
+            'macaron',          # Can not support macaron while injecting adapters at each 'f' layer
+        ):
+            if unsupported_key in opts_dict:
+                raise ValueError(
+                    f'"{unsupported_key}" is not supported in Mammoth.'
+                )
+        # Mammoth has a different default value than x-transformers,
```

Comment:
needs to be in the docs somewhere — perhaps in the help of the corresponding flag in opts?